pty/pty_other.go

```diff
@@ -66,3 +75,21 @@ func (p *otherPty) Close() error {
 	}
 	return nil
 }
+
+func (p *otherPtyWithProcess) Wait() error {
+	<-p.cmdDone
+	return p.cmdErr
+}
+
+func (p *otherPtyWithProcess) Kill() error {
+	return p.cmd.Process.Kill()
+}
+
+func (p *otherPtyWithProcess) waitInternal() {
+	// The GC can garbage collect the TTY FD before the command
+	// has finished running. See:
+	// https://github.com/creack/pty/issues/127#issuecomment-932764012
+	p.cmdErr = p.cmd.Wait()
+	runtime.KeepAlive(p.pty)
```

Comment:
TIL about [this](https://pkg.go.dev/runtime#KeepAlive), which doesn't exactly do what you would expect based on the name alone:
> KeepAlive marks its argument as currently reachable. This ensures that the object is not freed, and its finalizer is not run, before the point in the program where KeepAlive is called.

So, this does _not_ mean that the PTY is going to be kept alive forever.

Reply:
Yep, just until the underlying command process is done.